src/ring_theory/polynomial/homogeneous.lean

```diff
@@ -0,0 +1,160 @@
+/-
+Copyleft 2020 Johan Commelin. No rights reserved.
+Authors: Johan Commelin
+-/
+
+import data.mv_polynomial
+import data.fintype.card
+
+/-!
+# Homogeneous polynomials
+
+A multivariate polynomial `φ` is homogeneous of degree `n`
+if all monomials occuring in `φ` have degree `n`.
+-/
+
+open_locale big_operators
+
+namespace mv_polynomial
+variables {σ : Type*} {τ : Type*} {R : Type*} {S : Type*}
+
+/-- A multivariate polynomial `φ` is homogeneous of degree `n`
+if all monomials occuring in `φ` have degree `n`. -/
+def is_homogeneous [comm_semiring R] (φ : mv_polynomial σ R) (n : ℕ) :=
+∀ ⦃d⦄, coeff d φ ≠ 0 → ∑ i in d.support, d i = n
```

Comment:
Surely there is a `total_degree` function somewhere for monomials? Having to write out the summation seems awkward.

Reply:
I only know of `total_degree` for `mv_polynomial`s. But I agree that this needs fixing.

Reply:
Needs fixing now? Or at some point? How about at least add a TODO here.

Reply:
Done.